classes/settings/admin_setting_configdate.php

```diff
@@ -0,0 +1,80 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * An admin setting for a date input.
+ *
+ * @package    local_sitsgradepush
+ * @copyright  2024 onwards University College London {@link https://www.ucl.ac.uk/}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @author     Alex Yeung <k.yeung@ucl.ac.uk>
+ */
+
+namespace local_sitsgradepush\settings;
+
+use admin_setting_configtext;
+
+/**
+ * An admin setting for a date input.
+ */
+class admin_setting_configdate extends admin_setting_configtext {
```

Comment:
what is this for?

Reply:
It implements a datepicker in a text config, but it's been removed now.
As discussed,  I added a field in the mapping table to distinguish which mappings could apply extensions.